pontoon/base/views.py

```diff
@@ -997,21 +996,25 @@ def save_user_name(request):
 
 @login_required(redirect_field_name='', login_url='/403')
 @require_POST
-def request_project(request, slug):
-    """Request project to be added to locale."""
-    try:
-        locale = request.POST['locale']
-    except MultiValueDictKeyError as e:
-        return HttpResponseBadRequest('Bad Request: {error}'.format(error=e))
-
-    project = get_object_or_404(Project, slug=slug)
+def request_projects(request, locale):
+    """Request projects to be added to locale."""
+    slug_list = request.POST.getlist('projects[]')
     locale = get_object_or_404(Locale, code__iexact=locale)
 
+    # Validate projects
+    project_list = Project.objects.available().filter(slug__in=slug_list)
+    if not project_list:
+        return HttpResponseBadRequest('Bad Request: Non-existent projects specified')
+
+    projects = ''
+    for project in project_list:
+        projects += '{project} ({slug})\n'.format(project=project.name, slug=project.slug)
+
     if settings.ADMINS[0][1] != '':
         EmailMessage(
```

Comment:
A little bit general question: what will happen if e.g. smtp fails? Can we recover a list of requested locale?